pkg/parser/parser.go

```diff
@@ -567,6 +570,17 @@ func (p *Parser) parseBlockWithEndTokens(scope *scope, endTokens map[lexer.Token
 }
 
 func (p *Parser) advance() {
+	p.advanceOne()
+	for p.cur.Type == lexer.WS {
+		p.advanceOne()
+	}
+	inc := 2
+	for p.peek.Type == lexer.WS {
+		p.peek = p.lookAt(p.pos + inc)
+	}
+}
+
+func (p *Parser) advanceOne() {
```

Comment:
Needs a better name.

Reply:
yeah, but I think the WS insensitive is the more common one so I'd prefer to reserve `advance` for it i think.... to be seen.